packages/odyssey-react/src/components/Box/Box.tsx

```diff
@@ -296,6 +300,31 @@ export interface BoxProps {
    * Content selection
    */
   userSelect?: "none" | "text" | "contain" | "all" | "auto";
+  /**
+   * Whether to apply the body text color class
+   * @default body
+   */
+  color?: "body" | false;
+  /**
+   * Whether to apply the normal font weight class
+   * @default regular
+   */
+  fontWeight?: "regular" | false;
+  /**
+   * Whether to apply the normal font style class
+   * @default normal
+   */
+  fontStyle?: "normal" | false;
+  /**
+   * Whether to apply the base font size class
+   * @default base
+   */
+  fontSize?: "base" | false;
+  /**
+   * Whether to apply the base line height class
+   * @default normal
+   */
+  lineHeight?: "normal" | false;
```

Comment:
I am curious, would an 'inherit' string type work here as opposed to `false` and omitting the style all together? I mentioned this in the call and thought it might be one way to allow the classes and styles applied from pseudo-attribute state in a parent (e.g. a button) to take affect for its children within Box.

Reply:
I like this approach because the type no longer needs to be polymorphic (all strings) and it also simplifies the class application logic below as there is no outcome that requires omitting the class entirely

Reply:
bumping this comment @andrewberg-okta do you have any thoughts to share here?

Reply:
unfortunately `inherit` will not take the value of a sibling class, it still overrides the value of that sibling and inherits from the parent.

Reply:
hmmm I think I understand the nuance here now. ultimately the pseudo-class enabled styles from Button will end up on the same node as these styles so `inherit` won't make sense. If we were instead wrapping Box within button in some fashion `inherit` would work but in that case we'd prefer to have `button` element as the root. You may be able to structure the component that way in fact, use an instinsic button root but apply <Box> as a child span to get default styles and inherited styling etc.

Reply:
I'm not sure `inherit` will be safe generally. For this `<Button>` proposal, it works because we'd be assigning all of the props on `.root`, but for components with `<Box>` as root, they'll be exposed to any parent properties unless we specifically set them. That would include any parent styles in the DOM.

In the global CSS setup, most of these properties would go onto `html`, and then our reset would create inheritance on *elements* where appropriate. That way, classed styles always override the defaults, but the defaults are still universal. 

I'm wondering if there's a way to get these universal defaults back to a lower specificity. That'd resolve the Box/Component interplay. Another option would be pulling those styles via the module instead of React - `composes` or `uses` might work well here.

Reply:
@edburyenegren-okta @andrewberg-okta I wouldn't suggest we use `inherit` as a default, it would be an opt-in decision exactly like passing `false` to omit the style entirely is here. Someone would need to know what they're doing to use the API correctly which is fine.

In terms of ways to set the `.root` styling for box to a lower specificity (universal styling in your question above) this would work:

```
:where(.root) {
 ... default styles to be overriden when necessary by passing to Box via className ...
 color:
 font-family:
 font-size:
 line-height:
 ... etc ...
}
``` 

The where pseudo-class function will effectively make the specificity 0, allowing easy overrides from class selectors we pass into Box. It isn't supported in IE11 however which may not be a deal breaker I'm not sure.


Taking a step back, I'm not sure getting into specificity and override style code across component boundaries is what we want. Its a step backward in some ways. We can use the "inherit" API I'm describing above and a nested <Box/> within a button element to get the effect we want, or use the polymorphic API shown here although I think it feels more clunky than I'd like in its current form.

Reply:
Whoa, I was totally unaware of `:where()`! Definitely using this in my web component playground...

After some reading, I wouldn't advise using it broadly, but `:where` seems tailor-made for the challenge we're seeing with Box. `<Box>` usage remains consistent at root, no additional API layer, no style risks introduced by `inherit`, and developers can write styles with an assumption that they'll get all the defaults/resets.

It would also allow us to bring in the `margin`, `padding`, and other "reset" styles that had to be removed previously. That'd get a lot of duplicative CSS out of our components.

Lack of IE11 support could be a deal breaker, but we aren't supporting it currently either. 😬😅

All that said, don't consider my feedback blocking. I'm aiming for the lowest cognitive overhead here, but understand if that's not the highest priority.

Reply:
@jeffbelser-okta Yes the use of a child element will work technically, but comes with the associated gotchas of adding additional DOM elements. 
 - The correct element must be used (in the case of button specifically, non interactive phrasing content)
 - any css or interactivity that assumed a particular DOM structure needs to be refactored, page weight and render time increase, etc...
- finally `line-height`. If the parent is a block level element that only accepts phrasing content unwanted space could be introduced, and as specific rule written for the parent to get matching line-height

Reply:
@andrewberg-okta good points! Some of these feel like things we could easily overcome but in either the `false` or `inherit` case we're going to be expanding the API here in a similar fashion, perhaps not to the same degree. I'm ok with the polymorphic API it gives us what we need for Button to work the way we want.